src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -980,7 +975,7 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
             else if (point.Properties().IsRightButtonPressed())
             {
                 // CopyOnSelect right click always pastes
-                if (_terminal->IsCopyOnSelectActive() || !_terminal->IsSelectionActive())
+                if (!_selectionUpdatedThisCycle || !_terminal->IsSelectionActive())

```

Comment:
I'm having a bit of trouble understanding why the change in this if statement is still equivalent to the comment above of "CopyOnSelect right click always pastes", can you explain?

Reply:
Iiiii might have forgotten to push the commit that changed the condition here back to CoS 😅